aten/src/ATen/core/ivalue.h

```diff
@@ -872,17 +881,22 @@ struct TORCH_API IValue final {
 
   // Detect aliased tensors.
   struct HashAliasedIValue {
+    size_t hashTensor(const at::Tensor& ten) const {
+      if (ten.is_mkldnn()) {
+        // MKLDNN tensors dont have storage and dont create views
+        // or aliasing so we can just use Tensor pointer, TODO: find way
+        // to use mkldnn storage
+        return reinterpret_cast<size_t>(ten.unsafeGetTensorImpl());
+      } else if (ten.is_sparse()) {
+        return hashTensor(ten._values());
```

Comment:
so, our options are:
- hash on just index/values
this makes it so we will detect aliasing overlap between sparse tensors and non-sparse tensors only of one of the two.  
- maintain separate hashes for index and value... i think this is probably not worth it for what is the extremely rare (non-existant ?) case of mutating after freezing sparse tensors indices that also happens to be a tensor that is used elsewhere
- 
i think current approach is good, maybe we can  add comment re-ignoring indices